libcodechecker/server/session_manager.py

```diff
@@ -258,6 +280,32 @@ def __try_auth_root(self, auth_string):
 
         return False
 
+    def __try_auth_token(self, auth_string):
+        if not self.__database_connection:
+            return None
+
+        user_name, token = auth_string.split(':')
+
+        try:
+            # Try the database, if it is connected.
+            transaction = self.__database_connection()
+            auth_session = transaction.query(SessionRecord.token) \
+                .filter(SessionRecord.user_name == user_name) \
+                .filter(SessionRecord.token == token) \
+                .filter(SessionRecord.can_expire.is_(False)) \
+                .limit(1).one_or_none()
+
+            if not auth_session:
+                return False
+
+            return auth_session
+        except Exception as e:
+            LOG.error("Couldn't check login in the database: ")
+            LOG.error(str(e))
+        finally:
+            if transaction:
```

Comment:
If `self.__database_connection()` throws an exception for some reason `transaction` will not be set as a variable and the `finally` section will fail with unknown variable.
It wold be better to set `transaction=None` before `try`.